File: Programmers/Level 2/테이블 해시 함수/Solution.java
// 람다를 이용해서 자바로도 짜봄.

import java.util.*;

class Solution {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;
        
        Arrays.sort(data, (int[] d1, int[] d2) -> {
            if (d1[col - 1] != d2[col - 1]) return d1[col - 1] - d2[col - 1];
            else return d2[0] - d1[0];
        });
        
        for (int i = row_begin; i <= row_end; i++) {
            int sum = 0;
           for (int d : data[i - 1]) {
                sum += d % i;
           }
            answer ^= sum;
        }
        
        return answer;
    }
}

File: Programmers/Level 2/테이블 해시 함수/Solution.kt
// 테이블 해시 함수
// https://school.programmers.co.kr/learn/courses/30/lessons/147354

// 구현

// 하라는데로 정렬해주고, xor 누적 해주면된다.

class Solution {
    fun solution(data: Array<IntArray>, col: Int, row_begin: Int, row_end: Int): Int {
        var answer: Int = 0 
        
        data.sortWith(compareBy<IntArray> { it[col - 1] }.thenByDescending { it[0] })
        
        for (i in row_begin .. row_end) {
            var sum = 0
            for (d in data[i - 1]) {
                sum += d.rem(i)
            }
            answer = answer xor sum
        }
        
        return answer
    }
}